Main.elm

```diff
@@ -4,28 +4,37 @@ import Time
 import Html
 import Html.Attributes
 import Html.Events
-import Json.Decode
-import Json.Encode
+import Json.Decode as Decode
+import Json.Encode as Encode
+import Kinto
 import Process
 import Task
 
 
+kintoServer =
+    "https://kinto.dev.mozaws.net/v1/"
+
+
+
 -- Model
 
 
 type
     Msg
-    -- Load stored data: GetData -> NewData
-    -- then on user input: UpdateContent -> Debounce (via TimeOut) -> setData -> DataSaved
+    -- Load stored data: GetData -> DataRetrieved
+    -- then on user input: UpdateContent -> Debounce (via TimeOut) -> encryptData (out port) -> DataEncrypted (in port) -> saveData -> DataSaved
     = NewEmail String
     | NewPassphrase String
-    | NewData (Maybe String)
     | UpdateContent String
     | NewError String
     | GetData
+    | DataRetrieved (Result Kinto.Error String)
     | Lock
-    | DataSaved String
-    | DataNotSaved String
+    | DataEncrypted String
```

Comment:
What is the difference between DataDecrypted and DataNotDecrypted? should we add comments or rename this to make it explicit?